saleor/account/models.py

```diff
@@ -162,6 +164,29 @@ class Meta:
             (AccountPermissions.MANAGE_STAFF.codename, "Manage staff."),
         )
 
+    def __init__(self, *args, **kwargs):
+        super(User, self).__init__(*args, **kwargs)
+        self._effective_permissions = None
+
+    @property
+    def effective_permissions(self) -> "QuerySet[Permission]":
+        if self._effective_permissions is None:
+            self._effective_permissions = get_permissions()
+            if not self.is_superuser:
+                self._effective_permissions = self._effective_permissions.filter(
+                    user=self
+                )
+                self._effective_permissions |= Permission.objects.filter(
+                    group__user=self
+                )
```

Comment:
Shouldn't that be `.filter(Q(user=self) | Q(group__user=self))`? It would be even better if it used a subquery with `exists` to eliminate the problem of duplicate permissions.